### DISCUSSION TITLE
need help in extension development

### DISCUSSION THREAD

**Discussion by swastik959** [microsoft/vscode-discussions#654] [Extension Development QnA] [answered]
This is a code which I wrote for a extension please check why when I am running it I am unable to see it in the command pallet : const vscode = require('vscode');
import { spawn } from 'child_process';
// This method is called when your extension is activated
// Your extension is activated the very first time the command is executed

/**
 * @param {vscode.ExtensionContext} context
 */
 function activate(context) {

	// Use the console to output diagnostic information (console.log) and errors (console.error)
	// This line of code will only be executed once when your extension is activated
  let disposable = vscode.commands.registerCommand("extension.startVelaVet", () => {
    const activeEditor = vscode.window.activeTextEditor;
    if (activeEditor) {
      const fileName = activeEditor.document.fileName;
      const interval = 50000; // Interval in milliseconds (e.g., 50000 = 50 seconds)

      runVelaVetPeriodically(fileName, interval);
      vscode.window.showInformationMessage("Vela Vet started.");
    } else {
      vscode.window.showWarningMessage("No active editor found.");
    }
  });

  context.subscriptions.push(disposable);

context.subscriptions.push(disposable);
}
 function runVelaVetCommand(fileName) {
	const command = `vela def vet ${fileName}`;
	const process = spawn(command, { shell: true });
  
	process.stdout.on('data', (data) => {
	  // Handle the output of the command if needed
	  console.log(data.toString());
	});
  
	process.stderr.on('data', (data) => {
	  // Handle any error output if needed
	  console.error(data.toString());
	});
  
	process.on('close', (code) => {
	  // Handle the exit code if needed
	  console.log(`Child process exited with code ${code}`);
	});
  }
  function runVelaVetPeriodically(fileName, interval) {
	runVelaVetCommand(fileName);
  
	setInterval(() => {
	  runVelaVetCommand(fileName);
	}, interval);
  }
  

// This method is called when your extension is deactivated
function deactivate() {}

module.exports = {
	activate,
	deactivate
}User
Here is the json : 
  "name": "validation",
  "displayName": "validation",
  "description": "validation",
  "version": "0.0.1",
  "engines": {
    "vscode": "^1.79.0"
  },
  "categories": [
    "Other"
  ],
  "activationEvents": [
    "onCommand:extension.startVelaVet"
  ],
  "main": "./extension.js",
  "contributes": {
    "commands": [
      {
        "command": "extension.startVelaVet",
        "title": "startVelaVet"
      }
    ]
  },
  "scripts": {
    "lint": "eslint .",
    "pretest": "npm run lint",
    "test": "node ./test/runTest.js"
  },
  "devDependencies": {
    "@types/glob": "^8.1.0",
    "@types/mocha": "^10.0.1",
    "@types/node": "^20.2.5",
    "@types/vscode": "^1.79.0",
    "@vscode/test-electron": "^2.3.2",
    "eslint": "^8.42.0",
    "glob": "^8.1.0",
    "mocha": "^10.2.0",
    "typescript": "^5.1.3"
  }
}

**Comment by swastik959** [comment]
Also I am trying to develop a extension which will check syntax highlighting and check the diff between code present on a remote server and on my device such as git lens can you please guide me through it's process

**Comment by r3m0t** [comment, marked as answer]
Hi,

1. Use `Output: Focus on Output View` and go through every panel looking for relevant error messages.
2. Set breakpoints in your code and step through to see if your code is getting loaded and if it's erroring at any point.
3. Try starting from a sample extension from https://github.com/microsoft/vscode-extension-samples/, once it works slowly delete the included functionality and add your own, use `git commit -a` whenever it is still working.
4. It doesn't look like you're using Typescript, it can catch many issues in your code and is worth learning.
